mapboxgl/layers.py

```diff
@@ -0,0 +1,745 @@
+import json
+
+import numpy
+
+from mapboxgl.utils import color_map, height_map
+from mapboxgl import templates
+from mapboxgl.utils import img_encode, numeric_map
+
+
+class AbstractLayer():
+    """Abstract class for layers"""
+
+    def __init__(self,
+                 data,
+                 opacity=1,
+                 below_layer='',
+                 legend=True,
+                 legend_layout='vertical',
+                 legend_gradient=False,
+                 legend_text_numeric_precision=None,
+                 legend_key_shape='square',
+                 min_zoom=0,
+                 max_zoom=24,
```

Comment:
Also noting, we should maybe rename `min_zoom` and `max_zoom` because it can be confusing between `min_zoom`/`max_zoom` for the map instance. `layer_min_zoom` might be better